nameable.rb

```diff
@@ -0,0 +1,25 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+    Nameable.correct_name(@nameable)
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.strip
+  end
```

Comment:
- [ ] Kindly notice that the `correct_name` method of the `TrimmerDecorator` should be implemented like this.
```ruby
def correct_name
  @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
end
```